code/streaming/Hacl.Streaming.Blake2.fst

```diff
@@ -787,156 +730,76 @@ let blake2 (a : alg) (m : valid_m_spec a)
 /// Introducing intermediate definitions for the instantiation
 
 inline_for_extraction noextract
-let blake2s_32 (no_key : bool) (key_size : key_size_t Spec.Blake2S no_key) =
+let blake2s_32 =
   blake2 Spec.Blake2S M32 Blake2s32.blake2s_init Blake2s32.blake2s_update_multi
-         Blake2s32.blake2s_update_last Blake2s32.blake2s_finish no_key key_size 
+         Blake2s32.blake2s_update_last Blake2s32.blake2s_finish
 
 inline_for_extraction noextract
-let blake2b_32 (no_key : bool) (key_size : key_size_t Spec.Blake2B no_key) =
+let blake2b_32 =
   blake2 Spec.Blake2B M32 Blake2b32.blake2b_init Blake2b32.blake2b_update_multi
-         Blake2b32.blake2b_update_last Blake2b32.blake2b_finish no_key key_size
-
-inline_for_extraction noextract
-let optional_key_blake2s (no_key : bool) (key_size : key_size_t Spec.Blake2S no_key) =
-  (I.optional_key () I.Erased (k Spec.Blake2S no_key key_size))
+         Blake2b32.blake2b_update_last Blake2b32.blake2b_finish
 
 inline_for_extraction noextract
-let optional_key_blake2b (no_key : bool) (key_size : key_size_t Spec.Blake2B no_key) =
-  (I.optional_key () I.Erased (k Spec.Blake2B no_key key_size))
+let optional_key_blake2 = I.optional_key () I.Erased (I.stateful_unused unit)
```

Comment:
why is this optional here? you should just pick mode = `Erased` for blake2 if you're just giving up on the ability to use the key

Reply:
ah ok this is just a helper that probably saves you an inversion... ok

Reply:
Yes indeed, just a helper. I could inline it or maybe rename it to make it clearer. Would something like `unit_key` work?

Reply:
I actually just renamed it to `unit_key`